src/Workspaces/Core/MSBuild/MSBuild/Build/ProjectBuildManager.cs

```diff
@@ -186,7 +192,7 @@ public void EndBatchBuild()
             MSB.Execution.BuildManager.DefaultBuildManager.EndBuild();
 
             // unload project so collection will release global strings
-            _batchBuildProjectCollection.UnloadAllProjects();
+            _batchBuildProjectCollection?.UnloadAllProjects();
```

Comment:
Why a `?.` here? It feels like a Contract.ThrowIfNull() is more appropriate because we shouldn't have been able to get here without a build already going, right?

Reply:
yeah, I agree. I'll create a mop up PR

---
In reply to: [401259147](https://github.com/dotnet/roslyn/pull/42319#discussion_r401259147) [](ancestors = 401259147)